src/data/dataset_interface.py

```diff
@@ -0,0 +1,28 @@
+import torchvision.transforms as transforms
+
+from abc import ABC, abstractmethod
+
+class TaskDataset(ABC):
+    @abstractmethod
+    def __init__(self,root_dir, kernel_size, sigma, size, batch_size=4, num_workers=2):
+        self.root_dir = root_dir
+        self.kernel_size = kernel_size
+        self.sigma = sigma
+        self.size = size
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+        self.train_dataset = None
+        self.train_loader = None
```

Comment:
Can we add a component to further split the train into train and val? We will need both for training